### PR TITLE
feat: ERC1271 signature verification

### PR DESCRIPTION
**Pull request by fzavalia**

### CODE REVIEW

File: contracts/Rentals.sol [hunk 1]
```diff
@@ -563,9 +566,24 @@ contract Rentals is
...
-        require(signer == _offer.signer, "Rentals#_verifyOfferSigner: SIGNER_MISMATCH");
+    /// @dev Verify that the signature is valid for the provided signer and hash.
+    /// Will perform an ecrecover for EOA _signers and ERC1271 verification for contract _signers.
+    function _verifySigner(
+        address _signer,
+        bytes32 _hash,
+        bytes memory _signature
+    ) private view {
+        if (_signer.code.length == 0) {
+            require(_signer == ECDSAUpgradeable.recover(_hash, _signature), "Rentals#_verifySigner: SIGNER_MISMATCH");
```
Comment (nachomazzara) [inline contracts/Rentals.sol:580]: Should we need to verify if `signer != 0` as well?

Comment (fzavalia) [reply contracts/Rentals.sol:580]: when recovering the address from the signature + hash, the OZ library checks that. 
```
if (signer == address(0)) {
    return (address(0), RecoverError.InvalidSignature);
}
```

Comment (fzavalia) [reply contracts/Rentals.sol:580]:
```
function recover(
    bytes32 hash,
    uint8 v,
    bytes32 r,
    bytes32 s
) internal pure returns (address) {
    (address recovered, RecoverError error) = tryRecover(hash, v, r, s);
    _throwError(error);
    return recovered;
}
```

**Review by nachomazzara — APPROVED** [review]

File: test/Rentals.spec.ts [hunk 2]
```diff
@@ -3424,6 +3579,51 @@ describe('Rentals', () => {
       await estate.connect(lessor)['safeTransferFrom(address,address,uint256,bytes)'](lessor.address, rentals.address, estateId, bytes)
     })
 
+    it('should allow a lessor to accept the offer of a smart contract account. The smart contract account can then act as a tenant', async () => {
```
Comment (nachomazzara) [inline test/Rentals.spec.ts:3582]: What do you think of adding the next time to differentiate tests with the same description to add something like this:

```suggestion
    it('should allow a lessor to accept the offer of a smart contract account. The smart contract account can then act as a tenant' :: safeTransferFrom, async () => {
```

Comment (fzavalia) [reply test/Rentals.spec.ts:3582]: I think that differentiation is made by the "describe" that test is inside. different would be to have 2 different tests with the same name in the same place.

Comment (fzavalia) [reply test/Rentals.spec.ts:3582]: But I understand your point when trying to search for the test in the test file. 
Having 2 different matches can be confusing.

**Pull request merged**